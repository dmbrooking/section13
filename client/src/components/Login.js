import React from 'react'
import { Redirect } from 'react-router'
import {
  Container, Row, Col, Button, Form, FormGroup,
  Label, Input, Card, CardTitle, CardBody
} from 'reactstrap'

import '../styles.css'
import background from '../assets/background.jpg'

class Login extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      email: '',
      password: ''
    }
  }

  updateEmail = (event) => {
    this.setState({ email: event.target.value })
  }

  updatePassword = (event) => {
    this.setState({ password: event.target.value })
  }

  handleSubmit = (event) => {
    const { loginUser } = this.props

    event.preventDefault()
    loginUser(this.state.email, this.state.password)
  };

  render () {
    if (this.props.authenticated) return <Redirect to='/' />

    return (
      <Container className='vertical-center'>
        <img className='bg' src={background} alt='background' />
        <Row className='justify-content-sm-center'>
          <Col sm='8' md='7' lg='6' className='align-self-center'>
            <Card>
              <CardBody>
                <CardTitle className='text-center'>
                  <div>
                    <h4>Section 13</h4>
                  </div>
                </CardTitle>
                <br />
                <Form data-testid='loginForm' onSubmit={this.handleSubmit}>
                  <FormGroup>
                    <Label for='emailInput'>Email</Label>
                    <Input
                      type='email'
                      name='email'
                      id='emailInput'
                      placeholder='Enter your email address'
                      value={this.state.email}
                      onChange={this.updateEmail}
                    />
                  </FormGroup>
                  <FormGroup>
                    <Label for='passwordInput'>Password</Label>
                    <Input
                      type='password'
                      name='password'
                      id='passwordInput'
                      placeholder='Enter your password'
                      value={this.state.password}
                      onChange={this.updatePassword}
                    />
                  </FormGroup>
                  <FormGroup>
                    <Button type='submit' className='button button--blue'>
                      Log In
                    </Button>
                  </FormGroup>
                </Form>
              </CardBody>
            </Card>
          </Col>
        </Row>
      </Container>
    )
  }
}

export default Login
