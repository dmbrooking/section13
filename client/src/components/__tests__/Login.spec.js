import React from 'react'
import configureStore from 'redux-mock-store'

import Login from '../Login'

describe('<Login />', () => {
  const mockStore = configureStore()
  const loginUser = jest.fn()
  let store, wrapper

  beforeEach(() => {
    store = mockStore({ auth: { authenticated: false } })
    wrapper = shallow(<Login store={store} loginUser={loginUser} />)
  })

  it('matches saved snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('calls loginUser action when form is submitted', () => {
    wrapper.find('[data-testid="loginForm"]').simulate('submit', { preventDefault: () => { } })
    expect(loginUser).toHaveBeenCalledTimes(1)
  })
})
