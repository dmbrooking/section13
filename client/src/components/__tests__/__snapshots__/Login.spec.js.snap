// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> matches saved snapshot 1`] = `
<Container
  className="vertical-center"
  tag="div"
>
  <img
    alt="background"
    className="bg"
    src="background.jpg"
  />
  <Row
    className="justify-content-sm-center"
    tag="div"
  >
    <Col
      className="align-self-center"
      lg="6"
      md="7"
      sm="8"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <Card
        tag="div"
      >
        <CardBody
          tag="div"
        >
          <CardTitle
            className="text-center"
            tag="h5"
          >
            <div>
              <h4>
                Section 13
              </h4>
            </div>
          </CardTitle>
          <br />
          <Form
            data-testid="loginForm"
            onSubmit={[Function]}
            tag="form"
          >
            <FormGroup
              tag="div"
            >
              <Label
                for="emailInput"
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                Email
              </Label>
              <Input
                id="emailInput"
                name="email"
                onChange={[Function]}
                placeholder="Enter your email address"
                type="email"
                value=""
              />
            </FormGroup>
            <FormGroup
              tag="div"
            >
              <Label
                for="passwordInput"
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                Password
              </Label>
              <Input
                id="passwordInput"
                name="password"
                onChange={[Function]}
                placeholder="Enter your password"
                type="password"
                value=""
              />
            </FormGroup>
            <FormGroup
              tag="div"
            >
              <Button
                className="button button--blue"
                color="secondary"
                tag="button"
                type="submit"
              >
                Log In
              </Button>
            </FormGroup>
          </Form>
        </CardBody>
      </Card>
    </Col>
  </Row>
</Container>
`;
